src/tests/multihost/ad/test_dyndns.py

```diff
@@ -0,0 +1,312 @@
+    session_multihost.ad[0].run_command(cmd_zoneadd)
+    session_multihost.ad[0].run_command(cmd_zoneconfig)
+
+    def remove_extra_network():
+        """ Delete reverse zone"""
+        session_multihost.ad[0].run_command(f"dnscmd.exe /zonedelete {network_zone}.in-addr.arpa /f")
+
+    request.addfinalizer(remove_extra_network)
+
+
+@pytest.fixture(scope="function")
+def extra_interface(session_multihost, request):
+    """ Create extra interface """
+    dummy_interface = 'ibm' + str(random.randrange(99))
+    dummy_ip = '192.168.1.' + str(random.randrange(2, 255))
+    session_multihost.client[0].run_command('ip link add ' + dummy_interface + ' type dummy', raiseonerr=False)
+    session_multihost.client[0].run_command('ip addr add ' + dummy_ip + ' dev ' + dummy_interface, raiseonerr=False)
+
+    def remove_interface():
+        """ Delete interface """
+        session_multihost.client[0].run_commnd('ip addr flush dev ' + dummy_interface, raiseonerr=False)
+        session_multihost.client[0].run_command('ip link del ' + dummy_interface, raiseonerr=False)
+
+    request.addfinalizer(remove_interface)
+    return dummy_interface, dummy_ip
+
+
+@pytest.fixture(scope="function")
+def getptr(session_multihost, request):
+    """ Get network and ptr record """
+    ip = str(session_multihost.client[0].ip)
+    net_list = ip.split(".")
+    network = str(net_list[2] + '.' + net_list[1] + '.' + net_list[0])
+    ptr = str(net_list[3])
+
+    def removeptr():
+        """ Delete PTR record from DNS """
+        cmd_delete_ptr = f"dnscmd.exe /recorddelete {network}.in-addr.arpa {ptr}. PTR /f"
+        session_multihost.ad[0].run_command(cmd_delete_ptr, raiseonerr=False)
+    request.addfinalizer(removeptr)
+
+    return ip, network, ptr
+
+
+@pytest.mark.usefixtures("change_client_hostname")
+@pytest.mark.dyndns
+@pytest.mark.tier1_3
+class TestDynDns(object):
+
+    def test_0001_verify_with_default_setting(self, multihost, adjoin, getptr):
+        """
+        :title: IDM-SSSD-TC: ad_provider: dyndns: default settings
+        :id: 183cc040-0b8c-47af-8b19-8bcb72c3f001
+        :steps:
+          1. Join client to AD domain
+          2. Perform DNS lookup for hostname
+          3. Perform DNS lookup for IP
+        :expectedresults:
+          1. Client joins domain
+          2. Hostname resolves correctly
+          3. IP resolves correctly
+        """
+        adjoin(membersw='adcli')
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        hostname_cmd = multihost.client[0].run_command('hostname', raiseonerr=False)
+        hostname = hostname_cmd.stdout_text.rstrip()
+        (ip, network, ptr) = getptr
+        client.clear_sssd_cache()
+        multihost.client[0].run_command("cat /etc/sssd/sssd.conf", raiseonerr=False)
+        cmd_lookup = multihost.client[0].run_command(f"nslookup {hostname}", raiseonerr=False)
+        assert cmd_lookup.returncode == 0
```

Comment:
If you want to fail after the run_command, why not just drop the raiseonerr=False?  Was there a reason?  as it is now, if it fails on the assert, it still won't run the second nslookup run_command.   run_command in general will fail and raise an error and cause the test to fail.  Was that not happening for some reason?

Same basic question applies to any asserts checking for returncode == 0.

Reply:
I think that was mostly an artifact of testing and debugging. Removing the DNS forwarders in AD helped with the secondary and tertiary name servers in /etc/resolv.conf

Reply:
I'm ok with updating the assert returncode == 0 stuff to the next PR.